btree.go

```diff
@@ -64,6 +64,51 @@ type Item interface {
 	Less(than Item) bool
 }
 
+// FreeList represents a free list of btree nodes. By default each
+// BTree has its own FreeList, but multiple BTrees can share the same
+// FreeList.
+type FreeList struct {
+	freelist []*node
+	grow     bool
+}
+
+func NewFreeList() *FreeList {
+	return newFreeList(true)
+}
+
+func newDefaultFreeList() *FreeList {
+	return newFreeList(false)
+}
+
+func newFreeList(grow bool) *FreeList {
```

Comment:
The 'grow' bool concerns me here.  Principal of least surprise: I would expect these to be functionally equivalent, but thanks to 'grow', they're not:

    b1 := New(32)
    b2 := NewWithFreeList(32, NewFreeList())